doc/api/assert.md

```diff
@@ -6,6 +6,44 @@
 
 The `assert` module provides a simple set of assertion tests that can be used to
 test invariants.
+A `strict` and a `legacy` mode exist, while it is recommended to only use
+[`strict mode`][].
+
+## Strict mode
+
+When using the strict mode any assert function will use the equality used in the
+strict function mode. So [`assert.deepEqual`][] will for example work the same
+as [`assert.deepStrictEqual`][].
+
+It can be accessed using:
+
+```js
+const assert = require('assert').strict;
+```
+
+## Legacy mode
+
+> Stability: 0 - Deprecated: Use strict mode instead.
+
+When accessing `assert` directly instead of using the `strict` property the
+"non-strict" equality will be used for the "non-strict" function names e.g.
+[`assert.deepEqual`][].
+
+It can be accessed using:
+
+```js
+const assert = require('assert');
+```
+
+It is recommended to use the [ strict mode`][] instead as the loose equality
+checks are often not sufficient.
+
```

Comment:
Nit: They're called `non-strict`, `legacy`, and `loose` throughout the document. I'd suggest picking just one of those and sticking with it.

Nit: `are often not sufficient` might be better expressed as `can often have surprising results`.

Reply:
I rephrased it and hope it is clearer now :-)